app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +25,43 @@ def show
     }
   end
 
+  def raw
+    @version ||= version_collection.find(params[:id])
+
+    expires_in 1.year, public: true
+
+    if Archiver.external_archive_url?(@version.uri)
```

Comment:
JUST realized we have some old data with no `version.uri`. (Only ever happened with Versionista: in some rare cases it would store nothing, and in a very tiny number of cases we lost some old stored responses before we started archiving them ourselves, since it tops out at storing the 50 latest per page.)

Anyway. We should make sure `Archiver.external_archive_url?` and `FileStorage::<Whatever>.contains_url?` handle `nil` inputs and/or bail out early with a 404 if `@version.uri.nil?`.

Reply:
Alternatively, I guess we could allow an empty string for `version.uri` and migrate the DB to default this column to `""` and disallow NULLs. But having NULLs in the DB seems clearer to me. Dunno ¯\\\_(ツ)\_/¯

Reply:
Added a check for null URIs to the raw view. Definitely prefer to leave nulls in the database when they are a semantic reality of the data,